### PR TITLE
[Rust Server] Fix up model generation

### PR DESCRIPTION
**Pull request by richardwhiuk**
Changes:

- Correctly generate anyOf/oneOf models
- Fix up ToString / FromStr support for anyOf/oneOf models
- Disable PartialOrd generation for anyOf/oneOf models
- Generate models for inline enums, such as in anyOf/oneOf models, but also in a bunch of cases which weren't being handled correctly.
- Support enums in headers, and vectors of models in headers, which is used in the samples with enums.

As an example, this means instead of:

```yaml
    AnyOfObject:
      description: Test a model containing an anyOf
      anyOf:
      - type: string
        enum:
        - FOO
        - BAR
      - type: string
        description: Alternate option
```

resulting in:

```rust
pub struct AnyOfObject {}
```

it results in:

```rust
pub struct AnyOfObject(swagger::AnyOf2<models::AnyOfObjectAnyOf,String>);

pub enum AnyOfObjectAnyOf {
    #[serde(rename = "FOO")]
    Foo,
    #[serde(rename = "BAR")]
    Bar,
}
```

### PR checklist
 
- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [X] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [X] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Rust Technical Committee: @frol @farcaller @paladinzh @jacob-pro

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java [hunk 1]
```diff
@@ -993,6 +999,67 @@ public CodegenParameter fromRequestBody(RequestBody body, Set<String> imports, S
...
+        } else {
+            ref = null;
+        }
+
+        return ref;
+    }
+
+    @Override
+    public String getTypeDeclaration(Schema p) {
+        LOGGER.info("Getting type declaration for schema");
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java:1025]: should we change these LOGGER.info to LOGGER.debug instead?

**Comment by wing328** [comment]
> Generate models for inline enums, such as in anyOf/oneOf models, but also in a bunch of cases which weren't being handled correctly.


What about using the following to create inline enum models?

https://github.com/openapitools/openapi-generator/blob/master/docs/customization.md#inline-schema-naming


> RESOLVE_INLINE_ENUMS=true will refactor inline enum definitions into $ref

**Comment by richardwhiuk** [comment]
> > Generate models for inline enums, such as in anyOf/oneOf models, but also in a bunch of cases which weren't being handled correctly.
> 
> What about using the following to create inline enum models?
> 
> https://github.com/openapitools/openapi-generator/blob/master/docs/customization.md#inline-schema-naming
> 
> > RESOLVE_INLINE_ENUMS=true will refactor inline enum definitions into $ref

That's way I've used :)

https://github.com/OpenAPITools/openapi-generator/pull/19363/files#diff-a1ba55767c2157cd627ae82d8cb8d4433f31268ea70efa046964b5986a1d7660R124-R126

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java (hunk 1 above)
Comment (richardwhiuk) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java:1025]: Ah yes - possibly trace actually.

**Comment by wing328** [comment]
thanks for the pr

let's give it a try

**Pull request merged**